### PR TITLE
Universal fix for kernel upgrade on first boot

### PR DESCRIPTION
**Pull request by ecdye**
Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

File: build-image/first-boot.bash [hunk 1]
```diff
@@ -151,26 +149,29 @@ echo "OK"
...
+firmwareBefore="$(uname -r)"
 echo -n "$(timestamp) [openHABian] Updating repositories and upgrading installed packages... "
 apt-get install --fix-broken --yes &> /dev/null
 if [[ $(eval "$(apt-get --yes upgrade &> /dev/null)") -eq 100 ]]; then
   echo -n "CONTINUING... "
   dpkg --configure --pending &> /dev/null
   apt-get install --fix-broken --yes &> /dev/null
   if apt-get upgrade --yes &> /dev/null; then
-    if is_pi; then
+    if [[ $firmwareBefore != "$(uname -r)" ]]; then
```
Comment (mstormi) [inline build-image/first-boot.bash:160]: will it change the output of uname -r without a reboot ?

BTW why apt-get --yes upgrade twice ?

Comment (ecdye) [reply build-image/first-boot.bash:160]: Yes it will, because it will fail the first time if any packages are broken so it uses `dpkg` to fix any broken packages if it does fail and then completes the upgrade fully the second time. It is essentially a safety precaution that was implemented before I joined Dev and prevents weird edge case errors so I left it in.

Comment (mstormi) [reply build-image/first-boot.bash:160]: > Yes it will.

Did you test that on real HW ?
Sorry for being sceptic but better safe than sorry. I had to re-build the image three times until it worked properly, and you're about to change that piece of code now.

Comment (ecdye) [reply build-image/first-boot.bash:160]: It should, I will double check now.

Comment (ecdye) [reply build-image/first-boot.bash]: It didn't but I found a different solution.

File: build-image/first-boot.bash [hunk 2]
```diff
@@ -151,26 +149,29 @@ echo "OK"
...
+firmwareBefore="$(dpkg -s raspberrypi-kernel | grep "Version:[[:space:]]")"
 echo -n "$(timestamp) [openHABian] Updating repositories and upgrading installed packages... "
 apt-get install --fix-broken --yes &> /dev/null
 if [[ $(eval "$(apt-get --yes upgrade &> /dev/null)") -eq 100 ]]; then
   echo -n "CONTINUING... "
   dpkg --configure --pending &> /dev/null
   apt-get install --fix-broken --yes &> /dev/null
   if apt-get upgrade --yes &> /dev/null; then
-    if is_pi; then
+    if [[ $firmwareBefore != "$(dpkg -s raspberrypi-kernel | grep "Version:[[:space:]]")" ]]; then
```
[the remark this answers is not in the document]
Comment (ecdye) [reply build-image/first-boot.bash:160]: No it can't. Not to mention if that were to happen, the user would likely have a broken system anyway because of a incomplete kernel upgrade.

[the remark this answers is not in the document]
Comment (mstormi) [reply build-image/first-boot.bash:160]: ok then
